### PR TITLE
Create contact form

### PR DESCRIPTION
**Pull request by theKHutDeveloper**
This PR creates a simple contact form which consists of a text-area field and a submit button. There is validation on the form to make sure that an empty form can not be submitted.

Screen Shot 2019-06-27 at 15 33 37

### CODE REVIEW

File: app/views/pages/contact.html.erb [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
 <div class="govuk-grid-row dashboard-row">
+
   <div class="govuk-grid-column-two-thirds">
+    <%= form_tag(contact_path, method: :post, id: "contact_form") do %>
+      <div class="govuk-form-group">
+        <label class="govuk-label" for="more-detail">
+          You can use this form to ask a question, report a problem or suggest an improvement.
+        </label>
+        <span id="more-detail-hint" class="govuk-hint">
+          Do not include personal or financial information, like your National Insurance number or credit card details.
```
Comment (KPobeeNorris) [inline app/views/pages/contact.html.erb]: @theKHutDeveloper - would be worth asking Leigh to provide some content for this as this hint is pretty generic/not relevant to our users.

File: app/views/pages/contact.html.erb [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+            <span class="govuk-error-message">
+              The text box
+              <%= @contact.errors[:more_detail].first %>
+            </span>
+          <% end %>
+        <% end %>
+        <textarea class="govuk-textarea" id="more-detail" name="more_detail" rows="10" aria-describedby="more-detail-hint"></textarea>
+      </div>
+
+      <button class="gem-c-button govuk-button" type="submit">Submit</button>
```
Comment (KPobeeNorris) [inline app/views/pages/contact.html.erb]: @theKHutDeveloper - haven't come across this class for govuk-button before, can I ask what it does?

File: app/views/pages/contact.html.erb (hunk 1 above)
Comment (theKHutDeveloper) [reply app/views/pages/contact.html.erb]: Changed hint text following your advice

Screen Shot 2019-06-28 at 10 37 17

File: app/views/pages/contact.html.erb (hunk 2 above)
Comment (theKHutDeveloper) [reply app/views/pages/contact.html.erb]: Changed to the standard govuk button class

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -1,7 +1,30 @@
 # frozen_string_literal: true
 
 class PagesController < ApplicationController
+  include ActiveModel::Validations
```
Comment (rossjones) [inline app/controllers/pages_controller.rb]: Don’t think we need to do the validation in the controller so we? We can rely on the model to do the validation and generate the errors.

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -1,7 +1,30 @@
...
   def help; end
 
-  def contact; end
+  def contact
+    @contact = nil
+  end
+
+  def create_contact
+    @contact = Contact.new(contact_params[:more_detail])
+    @contact.valid?
```
Comment (rossjones) [inline app/controllers/pages_controller.rb]: Is this leftover? Is also used below

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (theKHutDeveloper) [reply app/controllers/pages_controller.rb]: Removed from controller

File: app/controllers/pages_controller.rb (hunk 4 above)
Comment (theKHutDeveloper) [reply app/controllers/pages_controller.rb]: Removed

**Pull request merged**